### PR TITLE
Added abstractions to OpenGL

### PR DESCRIPTION
**Pull request by WholesomeIsland**
# Summary of the PR
Added simple abstractions to core OpenGL functions/objects. also adds a new Example Called "Further Abstractions".

# Related issues, Discord discussions, or proposals
Links go here.

# Further Comments
What turned me away from OpenGL in the first place was the complicated API and everything you needed to do just to get a triangle. this PR tries to make it very simple to draw vertices with OpenGL. you would create a GLObject and pass the vertices, indices and a refrence to a GL object. the base Absrtactions used in the examples were used for the GLObject class.

```csharp
GLObject object = new GLObject(GL.GetApi(window), verts, indices);
void OnLoad(){
object.CreateResources("PathToVertexShader", "PathToFragmentShader");
}
void Render(double obj){
object.Render();
}
void OnClose(){
object.DisposeResources();
}
```
>Ignore the SHFB stuff, Its been removed.

### CODE REVIEW

**Comment by Perksey** [comment]
@RhysWhy you open to giving this a quick lookover?

File: src/OpenGL/Silk.NET.OpenGL.Abstactions/BufferObject.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Silk.NET.OpenGL.Abstactions
```
Comment (Perksey) [inline src/OpenGL/Silk.NET.OpenGL.Abstactions/BufferObject.cs:5]: Please delete this project, we don't want to provide our own abstractions package (yet)

**Review by Perksey — CHANGES_REQUESTED** [review]
Thanks so much for the pull request!

I have asked RhysWhy, the messiah of tutorials, to review the actual meat of this if he's happy to. I have a lot on my plate so it'll probably sit around for a while otherwise.

Once again please discuss these changes before you make them, it's much easier to tell you how to do things/what not to do before you've actually done it.

**Comment by RhysWhy** [comment]
@Perksey I took a look over the meat of this one as asked, just to give you a quick summary of my findings, the two main parts of this are as below:

a) Adding the Abstractions csproj with its corresponding classes
b) Adding a new Tutorial to the examples folder that relies on and demonstrates the objects added to the Abstraction csproj

As you've requested that the abstractions project be removed as its not (yet) desired then the additional example also would need removing, which is essentially this entire PR

**Comment by Perksey** [comment]
Gotcha, thanks for taking a look through anyway and apologies for delegating this to you, figured you knew the tutorials better than anyone and it's better than it sitting around for months until i can get some quality time to look through it.

@WholesomeIsland Per Rhys' findings, I'm going to close this pull request as we have our own plans to achieve something similar to this.

Please feel free to use either [the Discussions tab](https://github.com/Ultz/Silk.NET/discussions) or our Discord server to discuss these changes with us before making them so that we can tell you before you make the change whether we think it's something suitable for Silk.NET.

**Pull request closed without merging**